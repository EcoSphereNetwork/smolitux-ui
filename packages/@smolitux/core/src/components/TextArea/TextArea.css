.smolitux-textarea-container {
  width: 100%;
  position: relative;
}

.smolitux-textarea-container--full-width {
  width: 100%;
}

.smolitux-textarea-label {
  display: block;
  margin-bottom: 0.5rem;
  font-weight: 500;
  font-size: 0.875rem;
  color: var(--textarea-label-color, #374151);
}

.smolitux-textarea {
  display: block;
  width: 100%;
  padding: 0.625rem 0.75rem;
  font-size: 0.875rem;
  line-height: 1.5;
  color: var(--textarea-text-color, #1f2937);
  background-color: var(--textarea-bg, white);
  background-clip: padding-box;
  border: 1px solid var(--textarea-border-color, #d1d5db);
  border-radius: 0.375rem;
  transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  resize: vertical;
  min-height: 2.5rem;
}

.smolitux-textarea:focus {
  border-color: var(--textarea-focus-border-color, #3b82f6);
  outline: 0;
  box-shadow: 0 0 0 3px var(--textarea-focus-shadow-color, rgba(59, 130, 246, 0.3));
}

.smolitux-textarea::placeholder {
  color: var(--textarea-placeholder-color, #9ca3af);
  opacity: 1;
}

.smolitux-textarea:disabled,
.smolitux-textarea[readonly] {
  background-color: var(--textarea-disabled-bg, #f3f4f6);
  opacity: 0.7;
  cursor: not-allowed;
}

/* Sizes */
.smolitux-textarea--sm {
  padding: 0.375rem 0.5rem;
  font-size: 0.75rem;
  min-height: 2rem;
}

.smolitux-textarea--md {
  padding: 0.625rem 0.75rem;
  font-size: 0.875rem;
  min-height: 2.5rem;
}

.smolitux-textarea--lg {
  padding: 0.75rem 1rem;
  font-size: 1rem;
  min-height: 3rem;
}

/* Variants */
.smolitux-textarea--outline {
  background-color: var(--textarea-bg, white);
  border: 1px solid var(--textarea-border-color, #d1d5db);
}

.smolitux-textarea--filled {
  background-color: var(--textarea-filled-bg, #f3f4f6);
  border: 1px solid transparent;
}

.smolitux-textarea--filled:focus {
  background-color: var(--textarea-filled-focus-bg, #f9fafb);
}

.smolitux-textarea--unstyled {
  background-color: transparent;
  border: none;
  padding: 0;
  box-shadow: none;
}

.smolitux-textarea--unstyled:focus {
  box-shadow: none;
}

/* Error State */
.smolitux-textarea--error {
  border-color: var(--textarea-error-border-color, #ef4444);
}

.smolitux-textarea--error:focus {
  border-color: var(--textarea-error-border-color, #ef4444);
  box-shadow: 0 0 0 3px var(--textarea-error-shadow-color, rgba(239, 68, 68, 0.3));
}

.smolitux-textarea-error-text {
  color: var(--textarea-error-text-color, #ef4444);
  font-size: 0.75rem;
  margin-top: 0.25rem;
}

/* Helper Text */
.smolitux-textarea-helper-text {
  color: var(--textarea-helper-text-color, #6b7280);
  font-size: 0.75rem;
  margin-top: 0.25rem;
}

/* Character Count */
.smolitux-textarea-count {
  position: absolute;
  right: 0.5rem;
  bottom: 0.5rem;
  font-size: 0.75rem;
  color: var(--textarea-count-color, #6b7280);
  background-color: var(--textarea-count-bg, rgba(255, 255, 255, 0.8));
  padding: 0.125rem 0.25rem;
  border-radius: 0.25rem;
  pointer-events: none;
}

.smolitux-textarea-count--error {
  color: var(--textarea-count-error-color, #ef4444);
}

/* Auto-resize */
.smolitux-textarea--auto-resize {
  overflow: hidden;
}

/* Dark Mode */
.dark .smolitux-textarea-label {
  color: var(--textarea-label-color-dark, #e5e7eb);
}

.dark .smolitux-textarea {
  color: var(--textarea-text-color-dark, #f9fafb);
  background-color: var(--textarea-bg-dark, #1f2937);
  border-color: var(--textarea-border-color-dark, #4b5563);
}

.dark .smolitux-textarea:focus {
  border-color: var(--textarea-focus-border-color-dark, #60a5fa);
  box-shadow: 0 0 0 3px var(--textarea-focus-shadow-color-dark, rgba(96, 165, 250, 0.3));
}

.dark .smolitux-textarea::placeholder {
  color: var(--textarea-placeholder-color-dark, #6b7280);
}

.dark .smolitux-textarea:disabled,
.dark .smolitux-textarea[readonly] {
  background-color: var(--textarea-disabled-bg-dark, #374151);
}

.dark .smolitux-textarea--filled {
  background-color: var(--textarea-filled-bg-dark, #374151);
}

.dark .smolitux-textarea--filled:focus {
  background-color: var(--textarea-filled-focus-bg-dark, #4b5563);
}

.dark .smolitux-textarea--error {
  border-color: var(--textarea-error-border-color-dark, #f87171);
}

.dark .smolitux-textarea--error:focus {
  border-color: var(--textarea-error-border-color-dark, #f87171);
  box-shadow: 0 0 0 3px var(--textarea-error-shadow-color-dark, rgba(248, 113, 113, 0.3));
}

.dark .smolitux-textarea-error-text {
  color: var(--textarea-error-text-color-dark, #f87171);
}

.dark .smolitux-textarea-helper-text {
  color: var(--textarea-helper-text-color-dark, #9ca3af);
}

.dark .smolitux-textarea-count {
  color: var(--textarea-count-color-dark, #9ca3af);
  background-color: var(--textarea-count-bg-dark, rgba(31, 41, 55, 0.8));
}

.dark .smolitux-textarea-count--error {
  color: var(--textarea-count-error-color-dark, #f87171);
}