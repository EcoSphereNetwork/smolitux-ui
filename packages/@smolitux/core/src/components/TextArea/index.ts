export { TextArea } from './TextArea';
export type { TextAreaProps } from './TextArea';

// Fur Abwartskompatibilitat mit dem bestehenden Export
export { TextArea as default } from './TextArea';