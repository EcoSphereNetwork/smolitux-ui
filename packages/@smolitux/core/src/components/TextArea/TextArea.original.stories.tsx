import type { Meta, StoryObj } from '@storybook/react';
import { TextArea.original } from './TextArea.original';

const meta: Meta<typeof TextArea.original> = {
  title: 'Components/TextArea.original',
  component: TextArea.original,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'TextArea.original',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom TextArea.original',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive TextArea.original',
    onClick: () => alert('Clicked!'),
  },
};
