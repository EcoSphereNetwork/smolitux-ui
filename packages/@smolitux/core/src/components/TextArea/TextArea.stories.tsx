import type { Meta, StoryObj } from '@storybook/react';
import { TextArea } from './TextArea';

const meta: Meta<typeof TextArea> = {
  title: 'Components/TextArea',
  component: TextArea,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'TextArea',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom TextArea',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive TextArea',
    onClick: () => alert('Clicked!'),
  },
};
