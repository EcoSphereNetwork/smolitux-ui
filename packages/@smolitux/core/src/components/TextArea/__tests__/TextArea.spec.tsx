import React from 'react';
import { render } from '@testing-library/react';
import { TextArea } from '../TextArea';

// Mock fÃ¼r ThemeProvider
jest.mock('@smolitux/theme', () => ({
  useTheme: jest.fn(() => ({ themeMode: 'light' })),
}));

// Mock fÃ¼r FormControl
jest.mock('../../FormControl/FormControl', () => ({
  useFormControl: () => ({
    id: 'test-id',
    isDisabled: false,
    isInvalid: false,
    isReadOnly: false,
    isRequired: false,
  }),
}));

describe('TextArea Snapshots', () => {
  it('renders default textarea correctly', () => {
    const { asFragment } = render(<TextArea />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with label correctly', () => {
    const { asFragment } = render(<TextArea label="Description" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with helper text correctly', () => {
    const { asFragment } = render(<TextArea helperText="Enter a description" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with error message correctly', () => {
    const { asFragment } = render(<TextArea error="Description is required" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with placeholder correctly', () => {
    const { asFragment } = render(<TextArea placeholder="Enter description here" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with different sizes correctly', () => {
    const sizes = ['sm', 'md', 'lg'];
    
    const fragments = sizes.map(size => {
      const { asFragment } = render(<TextArea size={size as any} />);
      return { size, fragment: asFragment() };
    });
    
    fragments.forEach(({ size, fragment }) => {
      expect(fragment).toMatchSnapshot(`TextArea with size ${size}`);
    });
  });

  it('renders textarea with different variants correctly', () => {
    const variants = ['outline', 'filled', 'unstyled'];
    
    const fragments = variants.map(variant => {
      const { asFragment } = render(<TextArea variant={variant as any} />);
      return { variant, fragment: asFragment() };
    });
    
    fragments.forEach(({ variant, fragment }) => {
      expect(fragment).toMatchSnapshot(`TextArea with variant ${variant}`);
    });
  });

  it('renders textarea with fullWidth correctly', () => {
    const { asFragment } = render(<TextArea fullWidth />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with specified number of rows correctly', () => {
    const { asFragment } = render(<TextArea rows={5} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with maxLength correctly', () => {
    const { asFragment } = render(<TextArea maxLength={100} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with character count correctly', () => {
    const { asFragment } = render(<TextArea showCount maxLength={100} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with autoResize correctly', () => {
    const { asFragment } = render(<TextArea autoResize />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with value correctly', () => {
    const { asFragment } = render(<TextArea value="Sample text" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders disabled textarea correctly', () => {
    const { asFragment } = render(<TextArea disabled />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders readonly textarea correctly', () => {
    const { asFragment } = render(<TextArea readOnly />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders required textarea correctly', () => {
    const { asFragment } = render(<TextArea required />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with custom className correctly', () => {
    const { asFragment } = render(<TextArea className="custom-textarea" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with custom styles correctly', () => {
    const { asFragment } = render(<TextArea style={{ color: 'red' }} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders textarea with all features enabled correctly', () => {
    const { asFragment } = render(
      <TextArea 
        label="Description"
        helperText="Enter a description"
        placeholder="Enter description here"
        size="lg"
        variant="filled"
        fullWidth
        autoResize
        rows={5}
        maxLength={100}
        showCount
        className="custom-textarea"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});