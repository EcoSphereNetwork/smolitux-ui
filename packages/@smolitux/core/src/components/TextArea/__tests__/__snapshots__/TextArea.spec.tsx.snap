// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Snapshots renders default textarea correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders disabled textarea correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400 opacity-50 cursor-not-allowed "
      data-testid="textarea"
      disabled=""
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders readonly textarea correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      readonly=""
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders required textarea correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      required=""
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with all features enabled correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full"
    data-testid="textarea-container"
  >
    <label
      class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1"
      for="test-id"
    >
      Description
    </label>
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-5 py-3 text-lg border-0 bg-gray-100 dark:bg-gray-800 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  custom-textarea"
      data-testid="textarea"
      id="test-id"
      maxlength="100"
      placeholder="Enter description here"
      rows="5"
      style="height: 0px;"
    />
    <div
      class="mt-1 flex justify-between text-sm"
    >
      <p
        class="text-gray-500 dark:text-gray-400"
        data-testid="textarea-helper"
      >
        Enter a description
      </p>
      <p
        class="text-gray-500 dark:text-gray-400 "
        data-testid="textarea-counter"
      >
        0/100
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with autoResize correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
      style="height: 0px;"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with character count correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      maxlength="100"
      rows="3"
    />
    <div
      class="mt-1 flex justify-between text-sm"
    >
      <p
        class="text-gray-500 dark:text-gray-400 ml-auto"
        data-testid="textarea-counter"
      >
        0/100
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with custom className correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  custom-textarea"
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with custom styles correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
      style="color: red;"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different sizes correctly: TextArea with size lg 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-5 py-3 text-lg border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different sizes correctly: TextArea with size md 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different sizes correctly: TextArea with size sm 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-3 py-1.5 text-sm border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different variants correctly: TextArea with variant filled 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border-0 bg-gray-100 dark:bg-gray-800 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different variants correctly: TextArea with variant outline 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with different variants correctly: TextArea with variant unstyled 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border-0 bg-transparent px-0 py-0 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with error message correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      aria-describedby="test-id-error"
      aria-invalid="true"
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 border-red-500 dark:border-red-400 focus:ring-red-500 focus:border-red-500  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
    <div
      class="mt-1 flex justify-between text-sm"
    >
      <p
        class="text-red-600 dark:text-red-400"
        data-testid="textarea-error"
      >
        Description is required
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with fullWidth correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full"
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with helper text correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
    <div
      class="mt-1 flex justify-between text-sm"
    >
      <p
        class="text-gray-500 dark:text-gray-400"
        data-testid="textarea-helper"
      >
        Enter a description
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with label correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <label
      class="block text-sm font-medium text-gray-700 dark:text-gray-300 mb-1"
      for="test-id"
    >
      Description
    </label>
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with maxLength correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      maxlength="100"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with placeholder correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      placeholder="Enter description here"
      rows="3"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with specified number of rows correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="5"
    />
  </div>
</DocumentFragment>
`;

exports[`TextArea Snapshots renders textarea with value correctly 1`] = `
<DocumentFragment>
  <div
    class=""
    data-testid="textarea-container"
  >
    <textarea
      class="block rounded-md focus:outline-none focus:ring-2 transition duration-150 ease-in-out appearance-none w-full text-gray-900 dark:text-white placeholder-gray-400 dark:placeholder-gray-500 px-4 py-2 text-base border border-gray-300 dark:border-gray-600 bg-white dark:bg-gray-700 focus:ring-primary-500 focus:border-primary-500 dark:focus:ring-primary-400 dark:focus:border-primary-400  "
      data-testid="textarea"
      id="test-id"
      rows="3"
    >
      Sample text
    </textarea>
  </div>
</DocumentFragment>
`;
